sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,220 @@
+= sycl_ext_oneapi_group_load_store_extended
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:language: {basebackend@docbook:c++:cpp}
+
+== Introduction
+
+IMPORTANT: This specification is a draft.
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 5. All references below to the "core SYCL specification" or to section numbers in the SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_local_memory.asciidoc[
+  sycl_ext_oneapi_local_memory].
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[
+  sycl_ext_oneapi_group_sort].
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties].
+
+== Current State
+
+This proposal extends existing one for group load/store: link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_group_load_store.asciidoc[sycl_ext_oneapi_group_load_store]. Existing APIs are focused on single value/sycl::vec shown below:
+
+[source,c++]
+----
+// Load API
+template <typename T, access::address_space Space, access::decorated IsDecorated>
+T load(const multi_ptr<T, Space, IsDecorated>* src);
+
+template <int N, typename T, access::address_space Space, access::decorated IsDecorated>
+vec<T, N> load(const multi_ptr<T, Space, IsDecorated> src);
+
+// Store API
+template <typename T, access::address_space Space, access::decorated IsDecorated>
+void store(multi_ptr<T, Space, IsDecorated> dst, const T& x);
+
+template <typename T, access::address_space Space, access::decorated IsDecorated>
+void store(multi_ptr<T, Space, IsDecorated> dst, const vec<T, N>& x);
+----
+
+== Extended Proposal
+
+The extended APIs provides the following features:
+
+* load/store with temporary memory buffer, which can be passed via GroupHelper such as `group_with_scratchpad` link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc#group-helper[group_with_scratchpad description from sort over group proposal]
```

Comment:
Should there be overloads that _don't_ require a temporary memory buffer?

Reply:
it can be just the case when instead of `GroupHelper` just a group object is passed, but there won't be performance benefits in this case, I assume

Reply:
I assumed that the memory provided by the `GroupHelper` would only be used if the data needed to be transposed, and only if the group is a work-group.  It seems like if the data doesn't need to be transposed it can be returned directly to the work-item (without going through intermediate memory), and if the group is a sub-group the transpose could use shuffles.

Reply:
Resolved; there are now overloads accepting just a `Group`.